src/account/account.entity.ts

```diff
@@ -1,161 +1,91 @@
 import { randomUUID } from 'node:crypto';
-import { BaseEntity } from '../core/base.entity';
 import { type CreatePostType, PostType } from '../post/post.entity';
-import type { Site } from '../site/site.service';
 
-export type PersistedAccount = Account & { id: number };
+export interface Account {
+    readonly id: number;
+    readonly uuid: string;
+    readonly username: string;
+    readonly name: string | null;
+    readonly bio: string | null;
+    readonly url: URL;
+    readonly avatarUrl: URL | null;
+    readonly bannerImageUrl: URL | null;
+    readonly apId: URL;
+    readonly apFollowers: URL | null;
+    readonly isInternal: boolean;
+    /**
+     * Returns a new Account instance which needs to be saved.
+     */
+    updateProfile(params: ProfileUpdateParams): Account;
+    /**
+     * @deprecated
+     */
+    getApIdForPost(post: { type: CreatePostType; uuid: string }): URL;
+}
 
-export interface AccountData {
-    id: number;
-    uuid: string | null;
+export interface AccountDraft {
```

Comment:
What about this just being `type AccountDraft = Omit<Account, 'id'>` so we don't have to list all of the account properties here?

Reply:
Ah ok, yeh i'm fine with that, i guess it's unlikely they are going to come out of sync without us knowing